packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -5,82 +5,172 @@ import {
 					ariaLabel: sprintf( actionLabel, label ),
 				};
 			} ) || []
 		);
-	}, [ navigationMenus ] );
+	}, [
+		currentTitle,
+		currentMenuId,
+		navigationMenus,
+		createNavigationMenuIsSuccess,
+		isNavigationMenuResolved,
+		hasResolvedNavigationMenus,
+	] );
 
 	const hasNavigationMenus = !! navigationMenus?.length;
 	const hasClassicMenus = !! classicMenus?.length;
 	const showNavigationMenus = !! canSwitchNavigationMenu;
 	const showClassicMenus = !! canUserCreateNavigationMenu;
-	const hasManagePermissions =
-		canUserCreateNavigationMenu || canUserUpdateNavigationMenu;
-
-	// Show the selector if:
-	// - has switch or create permissions and there are block or classic menus.
-	// - user has create or update permissions and component should show the menu actions.
-	const showSelectMenus =
-		( ( canSwitchNavigationMenu || canUserCreateNavigationMenu ) &&
-			( hasNavigationMenus || hasClassicMenus ) ) ||
-		( hasManagePermissions && showManageActions );
-
-	if ( ! showSelectMenus ) {
-		return null;
+
+	useEffect( () => {
+		if ( ! hasResolvedNavigationMenus ) {
+			setSelectorLabel( __( 'Loading …' ) );
+		} else if ( hasNavigationMenus && ! currentMenuId ) {
+			setSelectorLabel( __( 'Select a menu' ) );
+			setEnableOptions(
+				( canSwitchNavigationMenu || canUserCreateNavigationMenu ) &&
+					hasResolvedNavigationMenus &&
+					( ! isCreatingMenu || createNavigationMenuIsSuccess )
+			);
+		} else if ( ! hasNavigationMenus && hasResolvedNavigationMenus ) {
+			setSelectorLabel( __( 'No menus. Create one?' ) );
+			setEnableOptions(
+				( canSwitchNavigationMenu || canUserCreateNavigationMenu ) &&
+					hasResolvedNavigationMenus &&
+					( ! isCreatingMenu || createNavigationMenuIsSuccess )
+			);
+		} else if ( hasResolvedNavigationMenus && currentMenuId === null ) {
+			setSelectorLabel( __( 'Select another menu' ) );
+			setEnableOptions(
+				( canSwitchNavigationMenu || canUserCreateNavigationMenu ) &&
+					hasResolvedNavigationMenus &&
+					( ! isCreatingMenu || createNavigationMenuIsSuccess )
+			);
+		}
+
+		if ( isCreatingMenu && createNavigationMenuIsSuccess ) {
+			setIsCreatingMenu( false );
+		}
+	}, [
+		currentMenuId,
+		hasNavigationMenus,
+		hasResolvedNavigationMenus,
+		createNavigationMenuIsSuccess,
+		isNavigationMenuResolved,
+	] );
+
+	toggleProps = {
+		...toggleProps,
+		className: 'wp-block-navigation__navigation-selector-button',
+		children: (
+			<>
+				<VisuallyHidden as="span">
+					{ __( 'Change Menu' ) }
+				</VisuallyHidden>
+				<Icon
+					icon={ isPressed ? chevronUp : chevronDown }
+					className="wp-block-navigation__navigation-selector-button__icon"
+				/>
+			</>
+		),
+		disabled: ! enableOptions,
+		onClick: () => {
+			setIsPressed( ! isPressed );
+		},
+	};
+
+	if ( ! hasNavigationMenus && ! hasClassicMenus ) {
+		return (
+			<Button
+				{ ...toggleProps }
+				className="wp-block-navigation__navigation-selector-button--createnew"
+				disabled={ ! enableOptions }
```

Comment:
`disabled` causes a focus loss, so I'd recommend against using it.

The `Button` has an `isBusy` for loading state. Return early from the `onClick` to prevent multiple clicks.

Reply:
> disabled causes a focus loss, so I'd recommend against using it.

TIL

Reply:
I have moved away from `disabled` to `isBusy`. Unfortunately preventing multiple clicks is more complicated.